### PR TITLE
Added support for custom length in cloudwatch components

### PR DESCRIPTION
**Pull request by thepalbi**
#### PR Description

The cloudwatch exporter components both in static and flow mode, expose a time model that allow the user to customize how much back in time to ask cloudwatch for metrics, and the aggregation period. Right now the only tunable param is `period` and both values are calculated based on that. This PR enables the user to tweak both `period` and `length` to custom values, falling back to the old period-based calculation when `length` is not configured.

#### Which issue(s) this PR fixes

Fixes issues below:

- https://github.com/grafana/support-escalations/issues/6519
- https://github.com/grafana/agent/issues/4285

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: docs/sources/flow/reference/components/prometheus.exporter.cloudwatch.md [hunk 1]
```diff
@@ -11,18 +11,25 @@ title: prometheus.exporter.cloudwatch
 
 # prometheus.exporter.cloudwatch
 
-The `prometheus.exporter.cloudwatch` component embeds [`yet-another-cloudwatch-exporter`](https://github.com/nerdswords/yet-another-cloudwatch-exporter). `` lets you collect [CloudWatch metrics](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/WhatIsCloudWatch.html), translate them to prometheus-compatible format and remote write.
+The `prometheus.exporter.cloudwatch` component
+embeds [`yet-another-cloudwatch-exporter`](https://github.com/nerdswords/yet-another-cloudwatch-exporter). `` lets you
```
Comment (marctc) [inline docs/sources/flow/reference/components/prometheus.exporter.cloudwatch.md:15]: this sentence ends with "`` lets you", which I think is wrong.

**Review by jcreixell — APPROVED** [review]

File: docs/sources/static/configuration/integrations/cloudwatch-exporter-config.md [hunk 2]
```diff
@@ -309,24 +309,41 @@ pick the ones you need.
...
 ```
 
-### Period
+### Period and length
+
+`period` controls primarily the width of the time bucket used for aggregating metrics collected from
+CloudWatch. `length`
+controls how far back in time CloudWatch metrics are considered during each agent scrape. If both settings are
+configured,
+the time parameters when calling CloudWatch APIs works as follows:
```
Comment (clayton-cornell) [inline docs/sources/static/configuration/integrations/cloudwatch-exporter-config.md:325]:
```suggestion
`period` controls the width of the time bucket used for aggregating metrics collected from CloudWatch.
`length` controls how far back in time CloudWatch metrics are considered during each agent scrape.
If both settings are configured, the time parameters when calling CloudWatch APIs work as follows:
```

Tidying up the line breaks

File: docs/sources/static/configuration/integrations/cloudwatch-exporter-config.md [hunk 3]
```diff
@@ -309,24 +309,41 @@ pick the ones you need.
...
+
+![](https://grafana.com/media/docs/agent/cloudwatch-period-and-length-time-model-2.png)
+
+As noted above, if across multiple metrics under the same static or discovery job, there's different `period`
+and/or `length`
+the minimum of all periods, and maximum of all lengths is configured.
```
Comment (clayton-cornell) [inline docs/sources/static/configuration/integrations/cloudwatch-exporter-config.md:331]:
```suggestion
As noted above, if there is a different `period` or `length` across multiple metrics under the same static or discovery job, 
the minimum of all periods, and maximum of all lengths is configured.
```

File: docs/sources/static/configuration/integrations/cloudwatch-exporter-config.md [hunk 4]
```diff
@@ -309,24 +309,41 @@ pick the ones you need.
...
 
-Period controls how far back in time CloudWatch metrics are considered, during each agent scrape. We can split how these
-settings affects the produced values in two different scenarios.
+On the other hand, if `length` is not configured, both period and length settings will be calculated based on the
+required
+`period` configuration attribute.
```
Comment (clayton-cornell) [inline docs/sources/static/configuration/integrations/cloudwatch-exporter-config.md:335]:
```suggestion
On the other hand, if `length` is not configured, both period and length settings are calculated based on 
the required `period` configuration attribute.
```

File: docs/sources/static/configuration/integrations/cloudwatch-exporter-config.md [hunk 5]
```diff
@@ -309,24 +309,41 @@ pick the ones you need.
...
 
-If all metrics within a job (discovery or static) have the same `Period` value configured, CloudWatch APIs will be requested
-for metrics from the scrape time, to `Periods` seconds in the past. The values of these are exported to Prometheus.
+If all metrics within a job (discovery or static) have the same `period` value configured, CloudWatch APIs will be
+requested
+for metrics from the scrape time, to `period`s seconds in the past. The values of these are exported to Prometheus.
```
Comment (clayton-cornell) [inline docs/sources/static/configuration/integrations/cloudwatch-exporter-config.md:339]:
```suggestion
requested for metrics from the scrape time, to `period`s seconds in the past. 
The values of these metrics are exported to Prometheus.
```

**Review by clayton-cornell — COMMENTED** [review]
Some line break tidying and a few minor suggestions

**Comment by thepalbi** [comment]
done @clayton-cornell

**Pull request merged**